app/views/unattended/provisioning_templates/snippet/remote_execution_ssh_keys.erb

```diff
@@ -46,7 +46,7 @@ if $user_exists; then
   mkdir -p <%= ssh_path %>
 
   cat << EOF >> <%= ssh_path %>/authorized_keys
-<%= host_param('remote_execution_ssh_keys').is_a?(String) ? host_param('remote_execution_ssh_keys') : host_param('remote_execution_ssh_keys').join("\n") %>
+<%= host_param('remote_execution_ssh_keys').is_a?(String) ? host_param('remote_execution_ssh_keys') : host_param('remote_execution_ssh_keys').uniq.join("\n") %>
```

Comment:
We are indeed calling `uniq` [here](https://github.com/theforeman/foreman_remote_execution/blob/484d107da45658139e830f67d66cf76c6f846022/app/models/concerns/foreman_remote_execution/host_extensions.rb#L86-L88), but that is too early. On my machine [this function](https://github.com/theforeman/foreman_remote_execution/blob/484d107da45658139e830f67d66cf76c6f846022/app/models/concerns/foreman_remote_execution/host_extensions.rb#L47) gets called 5 times and each call generates a single duplicate.

Either we should ensure that `host_params_hash` gets called only once or possibly add the uniq [here](https://github.com/theforeman/foreman_remote_execution/blob/484d107da45658139e830f67d66cf76c6f846022/app/models/concerns/foreman_remote_execution/host_extensions.rb#L53).

Either way, we should fix this in rex and not in the template

Reply:
in core we cache the result of `host_params_hash` (https://github.com/theforeman/foreman/blob/develop/app/models/concerns/host_params.rb#L41) to prevent duplicate calculations, but when rex overrides it it will append all the parameters every time the function is called. I agree this needs to be fixed in rex, but possibly it should be done by either changing the way the overriden function works or by hooking into a different step of the host parameters calculation process.